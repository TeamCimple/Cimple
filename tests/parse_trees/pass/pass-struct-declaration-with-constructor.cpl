struct MyStruct {
	int a;
	float b;
	string c;
	
	MyStruct(int new_a, float new_b, string new_c) {
		a = new_a;
		b = new_b;
		c = new_c;
	}

	~MyStruct() {

	}
};

int main() {
	return 0;
}
