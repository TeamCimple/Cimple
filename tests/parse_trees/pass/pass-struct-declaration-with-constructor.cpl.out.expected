Struct((int Var(a),), (float Var(b),), (string Var(c),), , MyStruct, Constructor(MyStructBody:
        CompoundStatement(Declarations:  StatementList: Statement(EQUALS(Identifier(a), Identifier(new_a))), Statement(EQUALS(Identifier(b), Identifier(new_b))), Statement(EQUALS(Identifier(c), Identifier(new_c))))),, )FuncDecl(Name: Var(main) ReturnType: int Parameters:  Body: CompoundStatement(Declarations:  StatementList: RETURN(Int(0)))) 
